### ISSUE TITLE
SGN vs. GJS Baselines Performance

### ISSUE THREAD

**Issue by JaWeyl** [ErikEnglesson/SGN#1]
Thanks for the excellent paper - I enjoyed reading it and the proposed method sounds exciting!

However, I've a question regarding the experimental evaluation - specifically, in comparision with your previous work about the [GJS](https://proceedings.neurips.cc/paper/2021/file/fe2d010308a6b3799a3d9c728ee74244-Paper.pdf) loss.

I compared the results for some of your proposed baselines on the CIFAR-10/100 datasets and found substantial differences in the achieved performance. Specifically, I'm a bit suprised by the difference in the CE baseline when using the training dataset in the _no noise_ setting, e.g., 64.87 (SGN paper) versus 77.60 (GJS paper) on CIFAR-100. Generally, in this setting all baselines in the GJS paper seem to achieve a substantial increased performance compared with the SGN paper.

Do you have any idea why this is the case? I am just curious about this since currently the comparsion between both papers is a bit hard for me.

Best,

JaWeyl

**Comment by ErikEnglesson** [comment]
Hi JaWeyl,

Thank you. I am glad you liked our new paper!

The GJS and SGN papers have very different implementations (note that this is in TensorFlow while the GJS is in PyTorch), and also important differences in the training setup:

- **Network Architecture**: SGN uses a WideResNet 28-2, while GJS uses a ResNet 34.
- **Data Augmentation**:  SGN uses simple data augmentation (random crop and flip), while GJS uses strong augmentations (RandAug). 

I believe both of these differences are important, but that the augmentation is the most important difference. In Table 5 in the GJS paper, we showed that the GJS method achieved an accuracy of 66.5% and 74.8% with weak and strong augmentations, respectively. In Table 8 of the SGN paper, we showed that stronger data augmentation in terms of mixup, had a significant performance improvement. I therefore believe stronger augmentations would improve the SGN performance on the CIFAR datasets too.

Note that the training setup we use in the SGN paper is very similar to the one in [Noise Against Noise](https://openreview.net/forum?id=80FMcTSZ6J0).

Due to the difference in training setup between SGN and GJS, I refrain from comparing numbers between the two papers. Unfortunately, I don't think it is easy to answer which method performs better than the other. Ideally, I would implement both in the same setup and compare that way.

Thanks again,
Erik

**Comment by JaWeyl** [comment]
Thanks for your prompt response and explaining the main differences between both papers - I appreciated it.

Cheers,

JaWeyl

**Comment by AndyWjk** [comment]
Hello @ErikEnglesson , I am wondering what's the reason for switching from Pytorch (GJS) to Tensorflow (SGN)? Is there any function that Pytorch does not support?

**Comment by ErikEnglesson** [comment]
Hi @AndyWjk,

Sorry for the late reply.

There were two main reasons why I switched from PyTorch to TensorFlow:
1. At the time I started the project, the TensorFlow Probability library was more mature than the corresponding PyTorch library (torch.distributions). For example, the multivariate normal distribution had an efficient low-rank parameterization of the covariance matrix, which did not exist in PyTorch.
2. Another reason was that I could save time on re-implementing baselines as some relevant related work had implementations in TensorFlow.

Best,
Erik
